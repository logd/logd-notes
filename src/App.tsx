import { Editor, EditorState } from 'draft-js';
import * as React from 'react';

class App extends React.Component<{}, any> {
  public onChange: any;
  constructor(props: any) {
    super(props);
    this.state = { editorState: EditorState.createEmpty() };
    this.onChange = (editorState: any) => this.setState({ editorState });
  }
  public render() {

    document.title = "Logd Note-Taking App";
    return (
      <div className="App">
        <h1>Draft.js Editor</h1>
        <div className="editor">
          <Editor
            editorState={this.state.editorState}
            onChange={this.onChange}
          />
        </div>
      </div>
    );
  }
}

export default App;
