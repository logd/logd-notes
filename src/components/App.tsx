import React, { useState, useEffect } from "react";
import "react-quill/dist/quill.snow.css";
import { Link, withRouter, RouteComponentProps } from "react-router-dom";
import { Routes } from "./Routes";
import { Auth } from "aws-amplify";

interface Props extends RouteComponentProps {

}

const AppComponent: React.FC<Props> = ({ location, history }) => {
  const [currentUser, setCurrentUser] = useState(null);
  const {pathname} = location;

  useEffect(() => {
    const updateCurrentUser = async () => {
      try {
        let user = await Auth.currentAuthenticatedUser();
        setCurrentUser({...user});
      } catch {
        setCurrentUser(null);
      }
    }
    updateCurrentUser();
  }, [])

  async function handleLogout(e: any) {
    e.preventDefault();
    try {
      await Auth.signOut();
      setCurrentUser(null);
      history.push("/login");
      
    } catch (error) {
      console.error(`Logout: ${error}`);
    }
  }

  return (
    <div>
      <div style={{
        padding: '10px 20px',
        display: 'flex' }}>
        <div style={{ flex: 1 }}><Link to="/">Logd</Link></div>
        <div style={{ flexBasis: '20%', textAlign: 'right' }}>
          {currentUser ?
            <button style={{
              // TODO: turn into LinkButton - https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md
              backgroundColor: 'transparent',
              border: 'none',
              cursor: 'pointer',
              textDecoration: 'underline',
              display: 'inline',
              margin: 0,
              padding: 0,
            }} 
            onClick={(e) => handleLogout(e)}>Sign out</button>
            :
            pathname === '/login' ?
            null
            :
            <Link to="/login">Login</Link>
          }
        </div>
      </div>
      <Routes appProps={{ currentUser, setCurrentUser }} />
    </div>
  );

}

export const App = withRouter(AppComponent);
