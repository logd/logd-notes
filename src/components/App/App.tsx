import React from "react";
import { AppHeader } from "../AppHeader/AppHeader";
import { Editor } from "../Editor/Editor";
import { Redirect, Route, Switch } from "react-router-dom";
import { Privacy, Home, Login } from "../../routes";
import styled from "styled-components";

const Wrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
`;

const Main = styled.div`
  flex: 1;
`;
export const App = () => (
  <Wrapper>
    <AppHeader />
    <Main>
      <Switch>
        <Route path="/" exact component={Home} />
        <Route path="/login" exact component={Login} />
        <Route path="/privacy" exact component={Privacy} />
        <Route path="/s/:space" component={Editor} />
        <Route
          path="*"
          render={() => (
            <Redirect
              to={{
                pathname: "/",
              }}
            />
          )}
        />
      </Switch>
    </Main>
  </Wrapper>
);
