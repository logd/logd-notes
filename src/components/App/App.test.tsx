import React from "react";
import renderer from "react-test-renderer";
import { App } from "./App";
import { MemoryRouter } from "react-router";

jest.mock("../Editor/Editor", () => {
  return {
    Editor: () => {
      return <></>;
    },
  };
});

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
