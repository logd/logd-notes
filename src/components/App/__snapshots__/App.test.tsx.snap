// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxhUy hUcYOB"
>
  <div
    className="sc-AxjAm eYjUwD"
  >
    <div
      className="sc-AxiKw efHbud"
    >
      logd
    </div>
    <div
      className="sc-AxirZ dudmGI"
    >
      Quick personal notes, saved locally only.
    </div>
  </div>
  <div
    className="sc-AxgMl cHYKOq"
  />
</div>
`;
